git/parse.go

```diff
@@ -80,6 +81,29 @@ func parseGPGSig(r *bufio.Reader, w *bytes.Buffer) error {
 	}
 }
 
+func parseTagGPGSig(r *bufio.Reader, w *bytes.Buffer) error {
+	//!Tag signatures do not have trailing whitespaces
+	for {
+		l, err := r.ReadString('\n')
+		if err != nil {
+			return nil
+		} else if !strings.Contains(l, "-----END PGP SIGNATURE-----") {
+			_, err = w.WriteString(fmt.Sprintf("\n%s", strings.Trim(l, " \n")))
+			if err != nil {
+				return err
+			}
+			continue
+		} else {
+			_, err = w.WriteString(fmt.Sprintf("\n%s", strings.Trim(l, " \n")))
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+
+		return fmt.Errorf("Unexpected end of gpg signature")
```

Comment:
Since the `if` block ends with an `else`, can we ever hit this line, or I am overlooking something?